### PR TITLE
Fix the display of utf8 sequences in the gui

### PR DESCRIPTION
**Pull request by LemonBoy**
term_addstr() had a long-standing fixme that suggested it didn't
take into account the string encoding when calculating the string
length.
The BIG5 code path is untested.

### CODE REVIEW

File: src/fe-text/term-terminfo.c [hunk 1]
```diff
@@ -522,15 +522,36 @@ void term_add_unichar(TERM_WINDOW *window, unichar chr)
...
+
+	len = 0;
+	raw_len = strlen(str);
+
+	/* The string length depends on the terminal encoding */
+
+	ptr = str;
+
+	if (term_type != TERM_TYPE_BIG5) {
+	    while (*ptr != '\0') {
```
Comment (dequis) [inline src/fe-text/term-terminfo.c]: 4 space indents

**Comment by dequis** [comment]
I heard on irc that one issue with this was that it didn't use wcwidth. Looks like it changed since then, this looks good to me.

File: src/fe-text/term-terminfo.c [hunk 2]
```diff
@@ -522,15 +522,36 @@ void term_add_unichar(TERM_WINDOW *window, unichar chr)
...
-	len = strlen(str); /* FIXME utf8 or big5 */
+
+	len = 0;
+	raw_len = strlen(str);
+
+	/* The string length depends on the terminal encoding */
+
+	ptr = str;
+
+	if (term_type != TERM_TYPE_BIG5) {
```
Comment (ailin-nemui) [inline src/fe-text/term-terminfo.c]: you must check for term_type == TERM_TYPE_UTF8 instead

**Comment by ailin-nemui** [comment]
lets see how this goeas

**Pull request merged**